src/extension-client/remix-extension.ts

```diff
@@ -16,8 +20,20 @@ export class RemixExtension<T extends Api = any> {
   protected currentRequest: PluginRequest
   public isLoaded = false
 
-  constructor() {
+  constructor(opts: RemixExtensionOptions) {
     window.addEventListener('message', event => this.getMessage(event), false)
+
```

Comment:
Maybe create a `defaultOptions(options)` method to define the default values : 
```typescript
function defaultOptions(options? : Partial<RemixExtensionOptions>): RemixExtensionOptions {
  const opts = options || {}
  return {
    useCustomBootStrapTheme: opts.useCustomBootStrapTheme || false
  } as RemixExtensionOptions
}
```
and use it in the constructor :
```typescript
const opts = defaultOptions(options)
this.initListenOnSwitchTheme(opts.useCustomBootStrapTheme)
```

Reply:
wouldn't defining this in the ctor work:
`constructor(options: RemixExtensionOptions = { useCustomBootStrapTheme: false }) {`